### PR TITLE
Dialog: Converting to mergeStyles part 1.2 - sub component file structure.

### PR DESCRIPTION
**Pull request by jordandrako**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Dialog also has sub-components I didn't know if I should have converted in the last PR.
- DialogContent converted to base + styled component.
- DialogFooter converted to base + styled component.
- Export new base components

### CODE REVIEW

File: common/changes/office-ui-fabric-react/mergeStyles-Dialog-round1_2018-02-27-19-41.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (mikewheaton) [inline common/changes/office-ui-fabric-react/mergeStyles-Dialog-round1_2018-02-27-19-41.json]: There should only be one change file per PR.

File: packages/office-ui-fabric-react/src/components/Dialog/index.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 export * from './Dialog';
 export * from './Dialog.base';
 export * from './DialogContent';
+export * from './DialogContent.base';
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Dialog/index.ts]: We're not going to export the base components for now, right?

Comment (jordandrako) [reply packages/office-ui-fabric-react/src/components/Dialog/index.ts]: I thought that was just for wrapper components where the style is required for the component functionality. I'll double check this one in particular.

Comment (jordandrako) [reply packages/office-ui-fabric-react/src/components/Dialog/index.ts]: Double checked with David and he still thinks we should expose them, but use inline styles in very specific cases like Brian's utility component.

**Pull request merged**